pkgs/applications/graphics/dosage/default.nix

```diff
@@ -1,18 +1,16 @@
-{ lib, python3Packages, fetchFromGitHub }:
+{ lib, python3Packages }:
 
 python3Packages.buildPythonApplication rec {
   pname = "dosage";
-  version = "2018.04.08";
-  PBR_VERSION = version;
+  version = "2.17";
 
-  src = fetchFromGitHub {
-    owner = "webcomics";
-    repo = "dosage";
-    rev = "b2fdc13feb65b93762928f7e99bac7b1b7b31591";
-    sha256 = "1p6vllqaf9s6crj47xqp97hkglch1kd4y8y4lxvzx3g2shhhk9hh";
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "0vmxgn9wd3j80hp4gr5iq06jrl4gryz5zgfdd2ah30d12sfcfig0";
   };
-  checkInputs = with python3Packages; [ pytest responses ];
-  propagatedBuildInputs = with python3Packages; [ colorama lxml requests pbr setuptools ];
+
+  checkInputs = with python3Packages; [ pytest pytest-xdist responses ];
```

Comment:
Please use pytestCheckHook.

Reply:
Done, not sure whether just including it in the checkInputs is enough.

I've also been thinking this expression might fit better under pythonPackages but I'm not sure what the policy here is.

Reply:
> Done, not sure whether just including it in the checkInputs is enough.

We try to get rid of this and to use the flags pytestCheckHook provides globally.

> I've also been thinking this expression might fit better under pythonPackages but I'm not sure what the policy here is.

End user programs should not be there. pythonPackages is meant for libraries used by programs or other libraries.